### PR TITLE
Bruce switch markdownlint

### PR DESCRIPTION
**Pull request by andrewbrucelau-zaber**
Switched to a more reliable markdown linter.

### CODE REVIEW

File: .github/workflows/check_examples.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ jobs:
       - id: inspect-examples
         name: Inspect Examples
         run: |
+          npm install markdownlint-cli2 --global
```
Comment (martinzak-zaber) [inline .github/workflows/check_examples.yml:28]: Create package.json with "npm init" in the project. Add by " npm install markdownlint-cli2". Then the instructions will be simply just "npm install".

Don't install anything globally. I'd think you know that from Python already.

File: tools/check_examples/README.md [hunk 2]
```diff
@@ -11,7 +11,11 @@ and called from `.github/workflows/check_examples.yml`
 
 ## Dependency
 
-The script uses `pdm` and the dependencies are listed in [`pyproject.toml`](pyproject.toml)
+The script uses `markdownlint-cli2` to lint markdown files.
+See [`markdownlint-cli2`](https://github.com/DavidAnson/markdownlint-cli2) for installation instruction.
```
Comment (martinzak-zaber) [inline tools/check_examples/README.md:15]: no no. Just "npm install". You can also then add "start" script that will run your linter.

File: .github/workflows/check_examples.yml (hunk 1 above)
Comment (martinzak-zaber) [reply .github/workflows/check_examples.yml:28]: That will make it reproducible by others.

File: tools/check_examples/src/check_examples/check_basic.py [hunk 3]
```diff
@@ -23,13 +23,8 @@ def check_markdown(directory: Path, recurse: bool = True) -> int:
...
     markdown_files = list_files_of_suffix(directory, ".md", recurse=recurse)
-    check_example_directory = get_git_root_directory() / "tools/check_examples"
-    config_file = check_example_directory / "pymarkdownlnt.toml"
     for file in markdown_files:
         filename = str(file)
-        return_code |= execute(
-            ["pdm", "run", "pymarkdownlnt", "--config", str(config_file.name), "scan", filename],
-            check_example_directory,
-        )
+        return_code |= execute(["markdownlint-cli2", filename], get_git_root_directory())
```
Comment (martinzak-zaber) [inline tools/check_examples/src/check_examples/check_basic.py:28]:
```suggestion
        return_code |= execute(["npx markdownlint-cli2", filename], get_git_root_directory())
```

or

```suggestion
        return_code |= execute(["npm run lint:md", filename], get_git_root_directory())
```
if you e.g. create "lint:md" npm script. Or "npm test" if you just call your script test. Anything that makes sense to you works.

File: .github/workflows/check_examples.yml (hunk 1 above)
Comment (andrewbrucelau-zaber) [reply .github/workflows/check_examples.yml]: Done

File: tools/check_examples/README.md (hunk 2 above)
Comment (andrewbrucelau-zaber) [reply tools/check_examples/README.md]: Done

File: tools/check_examples/src/check_examples/check_basic.py (hunk 3 above)
Comment (andrewbrucelau-zaber) [reply tools/check_examples/src/check_examples/check_basic.py]: Done

**Review by martinzak-zaber — APPROVED** [review]

**Pull request merged**